src/main/java/application/ServerLogger.java

```diff
@@ -0,0 +1,41 @@
+package application;
+
+import server.logger.ILogger;
+
+import java.io.File;
+import java.util.logging.FileHandler;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+
+public class ServerLogger implements ILogger {
+  private Logger logger;
+
+  public ServerLogger(String directoryLocation){
+    logger = Logger.getLogger(application.ServerLogger.class.getName());
+    makeLogDirectory(directoryLocation);
```

Comment:
This is a pretty big side effect to have in a constructor. I would move this into the `log` functions (so you can ensure it exists even if it's deleted after the logger has been instantiated). If it needs to happen up front, so you can add the handler to the logger, then I would move this into a static factory method with a name that implies some more complex initialization is happening. `ServerLogger.InitializeForDirectory(directoryLocation)`

Reply:
I see - I'll make some edits!